runtime/jvmti/jvmtiObject.c

```diff
@@ -206,59 +209,60 @@ jvmtiGetObjectMonitorUsage(jvmtiEnv* env,
 	TRACE_FOUR_JVMTI_RETURN(jvmtiGetObjectMonitorUsage2, rv_owner, rv_entry_count, rv_notify_waiter_count, rv_waiter_count);
 }
 
-
 static void
-findMonitorThreads(J9VMThread * vmThread, J9JVMTIMonitorStats * pStats)
+findMonitorThreads(J9VMThread *vmThread, J9JVMTIMonitorStats *pStats)
 {
-	j9object_t lockObject;
+	j9object_t lockObject = NULL;
+	J9VMThread *currentThread = (J9VMThread *)pStats->currentThread;
+	j9object_t target = (j9object_t)vmThread->threadObject;
+	UDATA threadState = 0;
+
+	if (NULL == target) {
+		return;
+	}
+
+#if JAVA_SPEC_VERSION >= 19
+	if (IS_JAVA_LANG_VIRTUALTHREAD(currentThread, target)) {
+		return;
+	}
+#endif /* JAVA_SPEC_VERSION >= 19 */
 
-	UDATA threadState = getVMThreadObjectStatesAll(vmThread, &lockObject, NULL, NULL);
+	threadState = getVMThreadObjectStatesAll(vmThread, &lockObject, NULL, NULL);
 
 	/* Stopped on the same monitor object? */
 	if (lockObject == pStats->lockObject) {
-
-		j9object_t target = (j9object_t)vmThread->threadObject;
-
-		if (target) {
-
-			J9JavaVM * vm = pStats->vm;
-			JNIEnv * jniEnv = (JNIEnv *)pStats->currentThread;
-
-			/* CMVC 87023 - the spec is unclear, but from experimentation it appears that 'waiting to be notified' threads
-			 * should appear in both lists
```

Comment:
Is this comment still relevant? Base on the JDK bug reports[[1]](https://bugs.openjdk.org/browse/JDK-8247972)&[[2]](https://bugs.openjdk.org/browse/JDK-8328083), this issue is only fixed for 23+. 
There is no backport to earlier versions and given that the problem is a mismatch between intended spec and actual implementation, Im not sure if we should have this affect the existing behaviour for JDK 8/11/17.

Reply:
In their [FIX](https://github.com/openjdk/jdk/pull/17680), they updated existing tests and mentioned of dependencies in the certification tests. If we propagate this fix to JDK22-, then existing tests might fail. We should enable these changes only in JDK23+, and follow the RI as they backport the fix to older JDK versions.

Reply:
Updated changes to only apply to JDK23+.